### PR TITLE
create new stripe customer at sign up

### LINKED ISSUE #87: BE: When a customer signs up, create a Stripe customer record
- When we sign up - create a customer record in Stripe. https://stripe.com/docs/api/customers/create?lang=node

### PR DESCRIPTION
**Pull request by cl-codes**
Resolves #87 

### What this PR does (required):
- Creates new stripe customer at sign-in (new back-end controller, route... API call... added to sign-up logic)
- Some other updates to clean up remnants of `user:name` in sign-up and demoSeed logic.

### Any information needed to test this feature (required):
- Sign up as a new user to check if new stripe customer ID is added to profile. Can also log into Stripe to see the same user (to double check the ID).

### CODE REVIEW

File: client/src/helpers/APICalls/stripe.ts [hunk 1]
```diff
@@ -13,4 +13,14 @@ const getLink = async (): Promise<StripeLink> => {
     }));
 };
 
-export default getLink;
+export const newCust = async (): Promise<string> => {
```
Comment (moffatethan) [inline client/src/helpers/APICalls/stripe.ts]: Let's make this `newCustomer`. More descriptive 🙂

File: client/src/pages/SignUp/SignUp.tsx [hunk 2]
```diff
@@ -32,6 +33,7 @@ export default function Register(): JSX.Element {
         updateSnackBarMessage(data.error.message);
       } else if (data.success) {
         updateLoginContext(data.success);
+        newCust();
```
Comment (moffatethan) [inline client/src/pages/SignUp/SignUp.tsx]: Couldn't we technically just do this logic on the back-end when they sign up?

**Review by moffatethan — CHANGES_REQUESTED** [review]
Just two comments - good work so far

File: client/src/pages/SignUp/SignUp.tsx (hunk 2 above)
Comment (cl-codes) [reply client/src/pages/SignUp/SignUp.tsx]: Latest version now has `auth/register` controller call stripe's `createCustomer` controller. Removed this logic from the front-end altogether.

File: client/src/helpers/APICalls/stripe.ts (hunk 1 above)
Comment (cl-codes) [reply client/src/helpers/APICalls/stripe.ts]: Got it - thanks! Ended up removing this API call now that logic is in the back-end.

**Review by moffatethan — APPROVED** [review]

**Pull request merged**